File: baekjoon/1004/1004.py
'''
어린 왕자 python 풀이
https://doctcoder.tistory.com/entry/%EC%96%B4%EB%A6%B0-%EC%99%95%EC%9E%90-%ED%92%80%EC%9D%B4
'''
t=int(input())
for i in range(t):
	x1,y1,x2,y2=map(int,input().split())
	n=int(input())
	x=[]
	y=[]
	r=[]
	cnt=0
	for j in range(n):
		cx,cy,cr=map(int,input().split())
		x.append(cx)
		y.append(cy)
		r.append(cr)
	for j in range(n):
		stadis=((x1-x[j])**2+(y1-y[j])**2)**0.5
		enddis=((x2-x[j])**2+(y2-y[j])**2)**0.5
		rad=r[j]
		if stadis<rad:
			if rad<enddis: cnt+=1
		elif rad>enddis: cnt+=1
	print(cnt)

File: baekjoon/1004/1004.cpp
/*
어린 왕자 C++ 풀이
https://doctcoder.tistory.com/entry/%EC%96%B4%EB%A6%B0-%EC%99%95%EC%9E%90-%ED%92%80%EC%9D%B4
*/
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main()
{
  int t;
	cin>>t;
	for(int i=0;i<t;i++)
	{
		int cnt=0;
		int x1,y1,x2,y2,n;
		cin>>x1>>y1>>x2>>y2;
		cin>>n;
		vector<int> cx(n);
		vector<int> cy(n);
		vector<int> cr(n);
		for(int j=0;j<n;j++) cin>>cx[j]>>cy[j]>>cr[j];
		for(int j=0;j<n;j++)
		{
			double sdis=pow(pow(x1-cx[j],2)+pow(y1-cy[j],2),0.5);
			double edis=pow(pow(x2-cx[j],2)+pow(y2-cy[j],2),0.5);
			double r=cr[j];
			if(sdis<r)
			{
				if(r<edis) cnt++;
			}
			else if(r>edis) cnt++;
		}
		cout<<cnt<<"\n";
	}
}
